scripts/lint-packages.js

```diff
@@ -106,29 +122,27 @@ for (let pkg of packages) {
 
 
 for (let pkg of packages) {
+  let globSrc = pkg.replace('package.json', '**/*.{js,ts,tsx}');
   let json = JSON.parse(fs.readFileSync(pkg));
   let [scope, basename] = json.name.split('/');
 
   if (basename.includes('utils') || basename.includes('layout')) {
     continue;
   }
-  if (scope === '@react-spectrum') {
-    let aria = `@react-aria/${basename}`;
-    let stately = `@react-stately/${basename}`;
-    let types = `@react-types/${basename}`;
 
+  let aria = `@react-aria/${basename}`;
+  let stately = `@react-stately/${basename}`;
+  let types = `@react-types/${basename}`;
+
+  if (scope === '@react-spectrum' && isDepUsed(aria, globSrc)) {
```

Comment:
Updated this lint step since it was complaining that I wasn't importing `react-stately/list` in `react-aria/list` even though `react-stately/list` wasn't used in `react-aria/list` at all. Now it checks that `@react-aria/${basename}`/`@react-stately/${basename}`/`@react-types/${basename}` is actually being imported somewhere in the package's files before asserting that it exists in the package.json.

Open to suggestions on the changes here and/or I've misunderstood the purpose of these checks

Reply:
dunno, I'd have to go through the aria packages we have to check if this is the first instance of this problem or not. If it isn't then I'm mystified why it was caught here specifically

Reply:
Ah right, so the reason why this wasn't a problem was because of this part of the check being passed to the `softAssert`: `!pkgNames[stately] || json.dependencies[stately]`. For a package like `@react-aria/accordion`, the first part will resolve to true since the `@react-stately/accordion` package doesn't exist (aka `pkgNames` doesn't contain `@react-stately/accordion`). However, for the newly introduced `@react-aria/list` package (ListView specific), `@react-stately/list` does exist so it expects it to be in the dependencies even though it isn't being used.

I could rename the listview package name to be `listview` instead of `list` which should side step this problem but it kinda breaks the previous naming conventions (e.g. TableView's package is `table`). Also I think that the update to this script still makes sense, doesn't feel like the previous logic was making a correct assumption.

Reply:
yep, thanks for looking into that! i think when I wrote that I assumed naming parity
so maybe we should still reconsider the name of the package, however, the change you made also makes sense